packages/flutter/test/widgets/sensitive_content_test.dart

```diff
@@ -0,0 +1,1414 @@
+          final DisposeTester nsc1 = DisposeTester(
+            child: SensitiveContent(
+              key: nsc1Key,
+              sensitivityLevel: ContentSensitivity.notSensitive,
+              child: Container(),
+            ),
+          );
+
+          await tester.pumpWidget(Column(children: <Widget>[sc1, asc1, nsc1]));
+
+          final DisposeTesterState nsc1DiposeTesterState = tester.firstState<DisposeTesterState>(
+            find.byKey(nsc1Key),
+          );
+          nsc1DiposeTesterState.disposeWidget();
+          await tester.pumpAndSettle();
+
+          expect(
+            sensitiveContentSetting.getContentSenstivityState()!.currentContentSensitivitySetting,
+            equals(ContentSensitivity.sensitive),
+          );
+          expect(
+            sensitiveContentSetting.getContentSenstivityState()!.sensitiveWidgetCount,
+            equals(1),
+          );
+          expect(
+            sensitiveContentSetting.getContentSenstivityState()!.autoSensitiveWidgetCount,
+            equals(1),
+          );
+          expect(
+            sensitiveContentSetting.getContentSenstivityState()!.notSensitiveWigetCount,
+            equals(0),
+          );
+          expect(setContentSensitivityCallCount, 1);
+        },
+      );
+
+      // Tests with another sensitive widget, an auto sensitive, and a not sensitive widget:
+      testWidgets(
+        'with another sensitive widget, one not sensitive widget, and one auto sensitive widget',
+        (WidgetTester tester) async {
+          final SensitiveContent sc1 = SensitiveContent(
+            sensitivityLevel: ContentSensitivity.sensitive,
+            child: Container(),
+          );
+          final SensitiveContent sc2 = SensitiveContent(
+            sensitivityLevel: ContentSensitivity.sensitive,
+            child: Container(),
+          );
+          final SensitiveContent asc1 = SensitiveContent(
+            sensitivityLevel: ContentSensitivity.autoSensitive,
+            child: Container(),
+          );
+          final SensitiveContent nsc1 = SensitiveContent(
+            sensitivityLevel: ContentSensitivity.notSensitive,
+            child: Container(),
+          );
+
+          await tester.pumpWidget(Column(children: <Widget>[sc1, sc2, asc1, nsc1]));
+
+          expect(
+            sensitiveContentSetting.getContentSenstivityState()!.currentContentSensitivitySetting,
+            equals(ContentSensitivity.sensitive),
+          );
+          expect(
+            sensitiveContentSetting.getContentSenstivityState()!.sensitiveWidgetCount,
+            equals(2),
+          );
+          expect(
+            sensitiveContentSetting.getContentSenstivityState()!.autoSensitiveWidgetCount,
+            equals(1),
+          );
+          expect(
+            sensitiveContentSetting.getContentSenstivityState()!.notSensitiveWigetCount,
+            equals(1),
+          );
+          expect(setContentSensitivityCallCount, 1);
+        },
+      );
+
```

Comment:
Super nit: Just as a thought to reduce lines of code in these tests, you could write a single test and put it in a loop over ContentSensitivity.values, and use that value as the sensitivity for the widget that gets disposed, or something like that.

Reply:
I actually prefer having the tests loop-less so it's clear the exact use case that each one is testing. I think reading the looped version could get kinda hard to understand.